packages/datagateway-download/src/downloadCart/downloadCartApi.ts

```diff
@@ -0,0 +1,190 @@
+import axios from 'axios';
+import * as log from 'loglevel';
+import { DownloadCart, DownloadCartItem, Datafile } from 'datagateway-common';
+
+// TODO: get URLs from settings or something...
+const topcatUrl = 'https://scigateway-preprod.esc.rl.ac.uk:8181/topcat';
+const apiUrl = 'http://scigateway-preprod.esc.rl.ac.uk:5000';
+
+export const fetchDownloadCartItems: () => Promise<DownloadCartItem[]> = () => {
+  return axios
+    .get<DownloadCart>(`${topcatUrl}/user/cart/LILS`, {
+      params: {
+        // TODO: get session ID from somewhere else (extract from JWT)
+        sessionId: window.localStorage.getItem('icat:token'),
+      },
+    })
+    .then(response => {
+      return response.data.cartItems;
+    })
+    .catch(error => {
+      log.error(error.message);
+      return [];
+    });
+};
+
+export const removeAllDownloadCartItems: () => Promise<void> = () => {
+  return axios
+    .delete(`${topcatUrl}/user/cart/LILS/cartItems`, {
+      params: {
+        // TODO: get session ID from somewhere else (extract from JWT)
+        sessionId: window.localStorage.getItem('icat:token'),
+        items: '*',
+      },
+    })
+    .then(() => {})
+    .catch(error => {
+      log.error(error.message);
+    });
+};
+
+export const removeDownloadCartItem: (
+  entityId: number,
+  entityType: string
+) => Promise<void> = (entityId: number, entityType: string) => {
+  return axios
+    .delete(`${topcatUrl}/user/cart/LILS/cartItems`, {
+      params: {
+        // TODO: get session ID from somewhere else (extract from JWT)
+        sessionId: window.localStorage.getItem('icat:token'),
+        items: `${entityType} ${entityId}`,
+      },
+    })
+    .then(() => {})
+    .catch(error => {
+      log.error(error.message);
+    });
+};
+
+// TODO: caching???
```

Comment:
Do you want this in with this commit?

Reply:
I've removed this comment, we think caching in TopCAT was a solution for when getSize calls were expensive - so we'll only look into caching if we observe performance issues.